zapc/src/hir_builder/decl.rs

```diff
@@ -59,148 +60,238 @@ impl<'a> HirBuilder<'a> {
 		}
 	}
 
-	fn event_config_from(&mut self, ast: AstConfigValue) -> HirEventSource {
+	fn event_config_from(&mut self, ast: AstConfigValue, event_name: &AstWord) -> HirEventSource {
 		match ast {
-			AstConfigValue::String(string) => {
-				match string.string(self.rodeo) {
-					"server" => HirEventSource::Server,
-					"client" => HirEventSource::Client,
-
-					other => {
-						// todo: report error
-						HirEventSource::Server
-					}
+			AstConfigValue::String(string) => match string.string(self.rodeo) {
+				"server" => HirEventSource::Server,
+				"client" => HirEventSource::Client,
+				_ => {
+					self.report(Report::ExpectedValue {
+						value_span: string.span(),
+						expected_values: "\"client\" or \"server\"".to_string(),
+						found: format!("\"{}\"", string.string(self.rodeo)),
+					});
+
+					HirEventSource::Server
 				}
-			}
+			},
 
 			AstConfigValue::Boolean(_, span) => {
-				// todo: report error
+				self.report(Report::ExpectedValueWrongType {
+					value_span: span,
+					expected_values: "\"client\" or \"server\"".to_string(),
+					found_type: "boolean".to_string(),
+					help: None,
+				});
+
 				HirEventSource::Server
 			}
 
 			AstConfigValue::Number(number) => {
-				// todo: report error
+				self.report(Report::ExpectedValueWrongType {
+					value_span: number.span(),
+					expected_values: "\"client\" or \"server\"".to_string(),
+					found_type: "number".to_string(),
+					help: None,
+				});
+
 				HirEventSource::Server
 			}
 
 			AstConfigValue::Path(words) => {
 				if words.len() == 1 {
 					let word = words.first().unwrap().word(self.rodeo);
 
-					if word == "server" || word == "Server" {
-						// todo: report special error
-					} else if word == "client" || word == "Client" {
-						// todo: report special error
+					if word == "server" || word == "client" {
+						self.report(Report::ExpectedValueWrongType {
+							value_span: words.first().unwrap().span().merge(words.last().unwrap().span()),
+							expected_values: "\"client\" or \"server\"".to_string(),
+							found_type: "path".to_string(),
+							help: Some("add quotes to turn the path into a string".to_string()),
+						});
+
+						return HirEventSource::Server;
+					} else if word == "Server" || word == "client" {
+						self.report(Report::ExpectedValueWrongType {
+							value_span: words.first().unwrap().span().merge(words.last().unwrap().span()),
+							expected_values: "\"client\" or \"server\"".to_string(),
+							found_type: "path".to_string(),
+							help: Some("lowercase the word and add quotes to turn the path into a string".to_string()),
+						});
+
+						return HirEventSource::Server;
 					}
-				} else {
-					// todo: report error
 				}
 
+				self.report(Report::ExpectedValueWrongType {
+					value_span: words.first().unwrap().span().merge(words.last().unwrap().span()),
+					expected_values: "\"client\" or \"server\"".to_string(),
+					found_type: "path".to_string(),
+					help: None,
+				});
+
```

Comment:
we will almost certainly not reuse `ExpectedValueWrongType`, so I think it would be a better idea to create a more specific report variant for this case.